controllers/submariner/submariner_controller_test.go

```diff
@@ -233,6 +234,7 @@ func testReconciliation() {
 
 	When("load balancer is enabled", func() {
 		BeforeEach(func() {
+			t.InitScopedClientObjs = append(t.InitScopedClientObjs, client.Object(newInfrastructureCluster()))
```

Comment:
@aswinsuryan @tpantelis , any idea how to address the unit testing failure?

I changed reconcileLoadBalancer function  to read Openshift ConfigV1().Infrastructures CR , unit test fails though I added infrastructure CR to t.InitScopedClientObjs

Reply:
It panics due to

`failed to add object &{{ } {cluster     999 0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] [] []} {{ } { <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil>}} { AWS <nil>      }} to fake client: no kind is registered for the type v1.Infrastructure in scheme "pkg/runtime/scheme.go:100"`

You need to register the type with the scheme. Add 

`Expect(v1config.Install(scheme.Scheme)).To(Succeed())`

after [this line](https://github.com/submariner-io/submariner-operator/blob/737cc2ccfd2cef23a9e87aa4bfaaabfc4c635c0c/controllers/submariner/submariner_suite_test.go#L48).

However, it then panics here:

```
runtime error: invalid memory address or nil pointer dereference

  Full Stack Trace
    github.com/openshift/client-go/config/clientset/versioned.NewForConfig(0x1e3c4b8?)
    	/home/tpanteli/go/src/github.com/submariner-io/submariner-operator/vendor/github.com/openshift/client-go/config/clientset/versioned/clientset.go:54 +0x29
    github.com/submariner-io/submariner-operator/controllers/submariner.(*Reconciler).getOCPPlatformType(0x18?, {0x1e39bb8, 0xc000192018})

```

```
func (r *Reconciler) getOCPPlatformType(ctx context.Context) string {
    configClient, err := configv1client.NewForConfig(r.config.RestConfig)
    ...
}

```
`r.config.RestConfig` is not  set by the unit tests since we can't create fake clients from a `rest.Config`. You need to use the `r.config.GeneralClient` to query the OpenShift resource. Also in the test setup you need to add it to `t.InitGeneralClientObjs`.

Reply:
The unit test panicked b/c `clusterInfra.Status.PlatformStatus` was nil - the test set the deprecated `Platform` instead. I pushed an [update](https://github.com/submariner-io/submariner-operator/compare/97b6d30d4d88293229ebf38709af07eb42a95d33..152a2257e6a56546b7959bd208d44c72c8ca590b) to fix this. In addition, I added unit test cases for _AWS_ and _IBMCloud_ and modified `getOCPPlatformType` to propagate the error.

Reply:
For the E2E failure, which you registered  the type with the scheme in _main.go_, the general client didn't use that scheme. I modified it to do so in [this update](https://github.com/submariner-io/submariner-operator/compare/ce29883fc683c2dfc1fb301b1e4e36f2f13bc2e0..5d51deafa593547d3c386803da17405548631de6).